### PR TITLE
Update action's description

### PR DESCRIPTION
**Pull request by ethomson**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-name: 'Cache'
-description: 'Cache dependencies and build outputs to improve workflow execution time'
+name: 'Cache Artifacts'
```
Comment (joshmgross) [inline action.yml]: @chrispat Do we want to change the name? I'm worried that "Artifacts" is overloaded and the docs specifically separate the two concepts: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows#comparing-artifacts-and-dependency-caching

As a user, "cache artifacts", "upload artifacts", and "download artifacts" sound like they should be very similar yet "cache" is fundamentally different in use and function.

Comment (ethomson) [reply action.yml]: We're not changing the name right now.

Comment (joshmgross) [reply action.yml]: @ethomson "Cache Artifacts" is what will show up in the Marketplace with the `action.yml` change to `name: 'Cache Artifacts'`

Comment (ethomson) [reply action.yml]: Sorry, to clarify: we're not going to merge this PR as-is.  We don't want a name change.

**Comment by joshmgross** [comment]
@ethomson and @chrispat, do we still want these description updates? And have we verified if an Action name update will break marketplace links?

**Comment by chrispat** [comment]
Yes we should make these changes and changing the name will not break the marketplace.

**Pull request merged**